src/Lykke.Service.PaySettlement.Services/AssetService.cs

```diff
@@ -0,0 +1,114 @@
+{
+    public class AssetService: IStartable, IAssetService
+    {
+        private readonly IAssetsService _assetsService;
+        private readonly ILog _log;
+        private readonly Core.Settings.AssetServiceSettings _settings;
+        private readonly Dictionary<string, AssetPair> _assetPairs;
+        private readonly Dictionary<string, Asset> _assets;
+
+        public AssetService(IAssetsService assetsService, ILogFactory logFactory,
+            Core.Settings.AssetServiceSettings settings)
+        {
+            _assetsService = assetsService;
+            _log = logFactory.CreateLog(this);
+            _settings = settings;
+            _assetPairs = new Dictionary<string, AssetPair>();
+            _assets = new Dictionary<string, Asset>();
+        }
+
+        public void Start()
+        {
+            var assetGetAllTask = _assetsService.AssetGetAllAsync();
+            var assetPairGetAllTask = _assetsService.AssetPairGetAllAsync();
+            Task.WhenAll(assetGetAllTask, assetPairGetAllTask).GetAwaiter().GetResult();
+
+            Asset[] assets = assetGetAllTask.Result.ToArray();
+            AssetPair[] assetPairs = assetPairGetAllTask.Result.Where(p =>
+                    string.Equals(p.BaseAssetId, _settings.PaymentAssetId, StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(p.QuotingAssetId, _settings.PaymentAssetId, StringComparison.OrdinalIgnoreCase))
+                .ToArray();
+
+            foreach (string settlementAssetId in _settings.SettlementAssetIds)
+            {
+                FillAsset(assets, settlementAssetId);
+                FillAssetPair(assetPairs, settlementAssetId);
+            }
+            FillAsset(assets, _settings.PaymentAssetId);
+        }
+
+        private void FillAsset(Asset[] assets, string assetId)
+        {
+            Asset asset = assets.FirstOrDefault(a =>
+                string.Equals(a.Id, assetId, StringComparison.OrdinalIgnoreCase));
+
+            if (asset == null)
+            {
+                _log.Critical(null, $"Asset {assetId} is not found.");
+            }
+
+            _assets[assetId.ToUpper()] = asset;
+        }
+
+        private void FillAssetPair(AssetPair[] assetPairs, string settlementAssetId)
+        {
+            AssetPair assetPair = assetPairs.FirstOrDefault(p =>
+                string.Equals(p.BaseAssetId, settlementAssetId, StringComparison.OrdinalIgnoreCase)
+                || string.Equals(p.QuotingAssetId, settlementAssetId, StringComparison.OrdinalIgnoreCase));
+
+            if (assetPair == null)
+            {
+                _log.Critical(null, $"AssetPair for assets {settlementAssetId} " +
+                                    $"and {_settings.PaymentAssetId} is not found.");
+            }
+
+            _assetPairs[settlementAssetId.ToUpper()] = assetPair;
+        }
+
+        public AssetPair GetAssetPair(string assetPairId)
+        {
+            return _assetPairs.Values.First(p =>
```

Comment:
I doubt this is performance-appropriate, but as this code will be deleted - nevermind.